code/modules/reagents/holder.dm

```diff
@@ -475,6 +475,7 @@
 			update_total()
 			if(my_atom)
 				my_atom.on_reagent_change(DEL_REAGENT)
+			SEND_SIGNAL(src, COMSIG_REAGENT_DELETED, reagent)
```

Comment:
you need to send to send signals before doing the action. So at beginning of the proc.

Also if you could fix that proc, it looks terrible 

no AS in, caching for no seemingly reason, isliving already returns false if the arg is null, my_atom?.on_reagent, return ture

Reply:
This is roughly mirroring the COMSIG_NEW_REAGENT_ADD, which's also at the end of the proc. In both cases, it's a better fit there because there's a bunch of verification that needs to happen and could end the proc without actually causing either signal event (adding a new reagent or removing an existing one). I can move it a few lines up, if you want? And yeah I'll take a peek at cleaning it up.